packages/manager/apps/catalog-revamp/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,205 @@
+  setSelectedUniverses: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const Filters: React.FC<FiltersProps> = ({
+  products,
+  setSelectedCategories,
+  setSelectedUniverses,
+}) => {
+  const [selectedCategories, setSelectedCategoriesLocal] = useState<string[]>(
+    [],
+  );
+  const [selectedUniverses, setSelectedUniversesLocal] = useState<string[]>([]);
+  const [hasInteracted, setHasInteracted] = useState<boolean>(false);
+
+  const setFilters = () => {
+    setSelectedCategories(selectedCategories);
+    setSelectedUniverses(selectedUniverses);
+  };
+
+  // we get the list of available categories and universes from product list
+  const universes = getUniversesWithCounter(products);
+  const categories = getAvailableCategoriesWithCounter(
+    products,
+    getUniverses(products),
+  );
+
+  const resetFilters = () => {
+    setSelectedCategoriesLocal([]);
+    setSelectedUniversesLocal([]);
+    setSelectedCategories([]);
+    setSelectedUniverses([]);
+  };
+
+  const { t } = useTranslation('catalog-revamp/filters');
+
+  const handleCheckboxChange = (
+    type: 'category' | 'universe',
+    label: string,
+    event: CustomEvent<{ checked: boolean; value: string }>,
+  ) => {
+    setHasInteracted(true);
+    const { checked } = event.detail;
+    if (type === 'category') {
+      setSelectedCategoriesLocal((prevSelected) =>
+        checked
+          ? [...prevSelected, label]
+          : prevSelected.filter((item) => item !== label),
+      );
+    } else {
+      setSelectedUniversesLocal((prevSelected) =>
+        checked
+          ? [...prevSelected, label]
+          : prevSelected.filter((item) => item !== label),
+      );
+    }
+  };
+
+  const FilterItem: React.FC<{
+    label?: string;
+    count?: number;
+    loading?: boolean;
+    type: 'category' | 'universe';
+  }> = ({ label, count, loading, type }) => {
+    const isChecked =
+      type === 'category'
+        ? selectedCategories.includes(label || '')
+        : selectedUniverses.includes(label || '');
+
+    const resultsNumber = count ? ` (${count})` : '';
+
+    return (
+      <div className="break-inside-avoid">
+        <OsdsCheckbox
+          checked={isChecked}
+          onOdsCheckedChange={(event: CustomEvent) =>
+            handleCheckboxChange(type, label, event)
+          }
+        >
+          <OsdsCheckboxButton
+            size={ODS_CHECKBOX_BUTTON_SIZE.sm}
+            color={ODS_THEME_COLOR_INTENT.primary}
+          >
+            <span slot="end">
+              {loading ? (
+                <OsdsSkeleton />
+              ) : (
+                <OsdsText
+                  color={ODS_THEME_COLOR_INTENT.default}
+                  level={ODS_TEXT_LEVEL.button}
+                  size={ODS_TEXT_SIZE._600}
+                >
+                  {label} {resultsNumber}
+                </OsdsText>
+              )}
+            </span>
+          </OsdsCheckboxButton>
+        </OsdsCheckbox>
+      </div>
+    );
+  };
```

Comment:
Avoid defining a component inside another component
That make it harder to read and maintain